### PR TITLE
Increase supported equil regions from 255 to 65525

### PR DESCRIPTION
**Pull request by blattms**
Previously we stored the equil region idx as unsigned short which imposed this limit. Apparently there models that use more regions than allowed. Hence we now store the idx using unsigned short which allows 65535 regions.

Still there is a caveat in the code: The region interaction is stored as a dense matrix. This means that using the maximum allowed number of regions now would consume 32 GB of memory. In the future we actually store the interactions in a sparse matrix to get rid of this memory requirement. For now we just issue a warning if there are more than 2048 regions used.

### CODE REVIEW

**Comment by blattms** [comment]
jenkins build this please

**Comment by akva2** [comment]
jenkins build this please

File: ebos/eclgenericthresholdpressure.cc [hunk 1]
```diff
@@ -133,6 +133,14 @@ finishInit()
                                maxRegions)));
     }
 
+    if (numEquilRegions_ > 2048) {
+        // warn about performance
+        OpmLog::warning(fmt::format("Number of equil regions is {}. This larger "
+                                 "than 2048. Note, that this might "
+                                 "might have a negative impact on performance "
+                                 "and memory consumption", numEquilRegions_));
+    }
```
Comment (akva2) [inline ebos/eclgenericthresholdpressure.cc:142]: read read and fix fix this message message please please. I'd just remove the 'This larger than 2048'. While you chose that as the warning limit, there is nothing significant about the particular value as such.

Comment (blattms) [reply ebos/eclgenericthresholdpressure.cc:143]: Thanks for the hint. Improved.

**Comment by blattms** [comment]
jenkins build this please

**Review by akva2 — APPROVED** [review]

**Pull request merged**